bin/run-example

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# make sure Spark home is set and valid
+if [ -z "${SPARK_HOME}" ]; then
+  echo "SPARK_HOME is not set" >&2
+  exit 1
+elif [ ! -d "${SPARK_HOME}" ]; then
+  echo "SPARK_HOME does not point to a valid directory" >&2
+  exit 1
+fi
+
+export _SPARK_CMD_USAGE="Usage: ./bin/run-example example-class [example args]"
+
+if (( $# == 0 )) || [[ $1 =~ "-h" ]]; then
+  echo "${_SPARK_CMD_USAGE}"
+  exit 1
+fi
+
+# split off example class from example arguments
+example_class="$1"; shift;
+example_args="$@"
+
+# pinpoint the project root folder
+bin_dir=$( dirname "$0" )
```

Comment:
minor suggestion: use `$(dirname "$(readlink -f "$0")")` since it also works in case the script is a symlink